File: docs/11.paradigm/answers/compose/trait.php
trait Walkable {
    public function walk() {
        echo "I am walking.\n";
    }
}

trait Swimmable {
    public function swim() {
        echo "I am swimming.\n";
    }
}

class Animal {
    public $name;
    public function __construct($name) {
        $this->name = $name;
    }
}

class Dog extends Animal {
    use Walkable, Swimmable;
}

$dog = new Dog("Buddy");
$dog->walk();   // 输出: I am walking.
$dog->swim();   // 输出: I am swimming.

File: docs/11.paradigm/answers/compose/trait.rs
trait Walkable {
   fn walk(&self);
}

trait Swimmable {
   fn swim(&self);
}

struct Dog {
   name: String,
}

impl Walkable for Dog {
   fn walk(&self) {
       println!("{} is walking.", self.name);
   }
}

impl Swimmable for Dog {
   fn swim(&self) {
       println!("{} is swimming.", self.name);
   }
}

fn main() {
   let dog = Dog { name: String::from("Buddy") };
   dog.walk();   // 输出: Buddy is walking.
   dog.swim();   // 输出: Buddy is swimming.
}